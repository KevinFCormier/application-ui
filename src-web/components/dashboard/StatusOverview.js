/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import lodash from 'lodash'
import PropTypes from 'prop-types'
import DashboardSection from '../common/DashboardSection'
import msgs from '../../../nls/platform.properties'
import { UtilizationCard } from '../UtilizationCard'
import {withRouter} from 'react-router-dom'

class StatusOverview extends React.Component {
  render() {
    const { locale } = this.context
    const { cardItems = [] } = this.props
    return (
      <div id='resource-utilization'>
        <DashboardSection name={msgs.get('dashboard.section.status-overview', locale)}>
          {cardItems.map(item =>
            item.name &&
            <UtilizationCard
              critical={item.critical}
              warning={item.warning}
              healthy={item.healthy}
              table={item.table}
              percentage={item.percentage}
              type={item.type}
              title={msgs.get(`dashboard.card.${lodash.camelCase(item.name)}`, locale)}
              key={item.name}
              id={`card-${item.name.replace(' ', '-')}`}
            />)}
        </DashboardSection>
      </div>
    )
  }
}

DashboardSection.contextTypes = {
  locale: PropTypes.string
}

StatusOverview.propTypes = {
  cardItems: PropTypes.arrayOf(PropTypes.shape({
    critical: PropTypes.number,
    healthy: PropTypes.number,
    warning: PropTypes.number,
    name: PropTypes.string,
  }))
}

export default withRouter(StatusOverview)
