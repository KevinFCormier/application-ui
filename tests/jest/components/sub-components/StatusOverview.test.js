/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import StatusOverview from '../../../../src-web/components/dashboard/StatusOverview'

describe('StatusOverview no components', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <StatusOverview />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

const mockItem = {
  critical:1,
  warning:2,
  healthy:3,
  name:'4'
}

describe('StatusOverview no components', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <StatusOverview cardItem={mockItem} />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
