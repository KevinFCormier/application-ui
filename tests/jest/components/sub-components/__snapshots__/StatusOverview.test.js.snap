// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusOverview no components renders as expected 1`] = `
<div
  id="resource-utilization"
>
  <div>
    <div
      className="dashboard-section-title"
    >
      <h2>
        Resource Utilization
      </h2>
    </div>
    <div
      className="dashboard-section-modules"
    />
  </div>
</div>
`;

exports[`StatusOverview no components renders as expected 2`] = `
<div
  id="resource-utilization"
>
  <div>
    <div
      className="dashboard-section-title"
    >
      <h2>
        Resource Utilization
      </h2>
    </div>
    <div
      className="dashboard-section-modules"
    />
  </div>
</div>
`;
